### PR TITLE
Add message to login screen if Caps Lock is on

### PR DESCRIPTION
**Pull request by bjarnef**
Issue: http://issues.umbraco.org/issue/U4-7362

I have used this module https://github.com/FabioMR/ng-caps-lock to add a message to login screen, when Caps Lock is on similar to the login on Windows.

However I kept getting a console error ```Uncaught Error: No module: ngCapsLock```.
First I tried with a static file in the lib folder, but think it should be installed with Bower as well? ```bower install ng-caps-lock```
Not should if the module should be added to /Umbraco.Web.UI.Client/lib/angular/ , as bower component or elsewhere?

The module ngCapsLock is injected a few places, but I must have missed something in how the module should be injected?

Let me now if I should made further changes and feel free to merge the changes here manually to test it.

### CODE REVIEW

File: src/Umbraco.Web.UI.Client/bower.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     "jquery-ui": "1.10.3",
     "angular-dynamic-locale": "~0.1.27",
     "tinymce": "~4.1.10",
-    "codemirror": "~5.3.0"
+    "codemirror": "~5.3.0",
+    "ngCapsLock": "~1.0.2"
```
Comment (Shazwazza) [inline src/Umbraco.Web.UI.Client/bower.json]: This seems incorrect, looks like it's supposed to be: ng-caps-lock

File: src/Umbraco.Web.UI.Client/src/loader.dev.js [hunk 2]
```diff
@@ -10,7 +10,9 @@ LazyLoad.js(
         'lib/angular/1.1.5/angular-mobile.js',
         'lib/angular/1.1.5/angular-sanitize.min.js',
 
-        'lib/angular/angular-ui-sortable.js',
+        'lib/angular/ng-caps-lock.js',
```
Comment (Shazwazza) [inline src/Umbraco.Web.UI.Client/src/loader.dev.js]: This is also required in /Umbraco.Web/UI/JavaScript/JsInitialize.js for c# server side.

File: src/Umbraco.Web.UI.Client/src/loader.dev.js [hunk 3]
```diff
@@ -10,7 +10,9 @@ LazyLoad.js(
...
+
+        'lib/ngCapsLock/angular-ui-sortable.js',
```
Comment (Shazwazza) [inline src/Umbraco.Web.UI.Client/src/loader.dev.js]: i think this is a typo ;)

**Comment by Shazwazza** [comment]
This is pulled in and tested but I had to fix quite a lot: 068c4a0b168818eb17ccf04a8cfe3fb876f38fe2

**Comment by bjarnef** [comment]
Thanks for reviewing it @Shazwazza .. it seems I missed some additions in a couple of files, so the grunt build + build.bat probably didn't include the Bower component at that time I was testing.

**Comment by bjarnef** [comment]
@Shazwazza I just tested it almost works as it should.

However I noticed some minor issues.
* On init if enter "caps lock" it doesn't change the "state" (show "caps lock on" message). It doesn't necessary have to show the message before something is entered into the form.

* If I enter username and with mouse place cursor into password field (or using tabs button) and tab "caps lock" it doesn't show the message and tabbing it again (off) then show the " Caps Lock is on" message (oppisite effect).

Here is a short capture of it: https://bjarnefyrstenborg.tinytake.com/sf/Mzk0NzQzXzIxOTkwODU

**Comment by Shazwazza** [comment]
As it stands now, the way it is now will be released. We do not have any more time to invest in this release. If you feel like it should be fixed to your liking then we will accept another PR by tomorrow latest. Otherwise if you think it's buggy then we can revert and not include it in this release at all. Let me know what you'd like to do.

**Comment by bjarnef** [comment]
It okay - just wanted to write a note about it.

Actually I also noticed that IE Edge (or I think IE10+) show a tooltip, but only for when hitting "caps lock" in password field.
But it seems to be possible to turn it off https://msdn.microsoft.com/en-us/library/jj128374%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396

image

So maybe revert it for the 7.3.2 release and we can test it further.

**Pull request merged**